external-table/src/pxfheaders.c

```diff
@@ -492,6 +509,163 @@ add_projection_desc_httpheader(CHURL_HEADERS headers,
 	pfree(formatter.data);
 	bms_free(attrs_used);
 }
+#endif
+
+/*
+ * Report projection description to the remote component, the indices of
+ * dropped columns do not get reported, as if they never existed, and
+ * column indices that follow dropped columns will be shifted by the number
+ * of dropped columns that precede it. For example,
+ *
+ *  ---------------------------------------------
+ * |  col1  |  col2 (dropped)  |  col3  |  col4  |
+ *  ---------------------------------------------
+ *
+ * Let's assume that col1 and col4 are projected, the reported projected
+ * indices will be 0, 2. This is because we use 0-based indexing and because
+ * col2 was dropped, the indices for col3 and col4 get shifted by -1.
+ */
+
+#if PG_VERSION_NUM >= 120000
+static void
+add_projection_desc_httpheader(CHURL_HEADERS headers,
+                 ProjectionInfo *projInfo,
+                 List *qualsAttributes,
+                 Relation rel)
+{
+  int        i;
+  int        dropped_count;
+  int        number;
+  int        numTargetList;
+  char      long_number[sizeof(int32) * 8];
+    // In versions < 120000, projInfo->pi_varNumbers contains atttribute numbers of SimpleVars
+    // Since,this pi_varNumbers doesn't exist in PG12 and above, we can add the attribute numbers by
+    // iterating on the Simple vars.
+  int       varNumbers[sizeof(int32) * 8];
+  Bitmapset   *attrs_used;
+  StringInfoData  formatter;
+  TupleDesc   tupdesc;
+  initStringInfo(&formatter);
+  numTargetList = 0;
+
+   List *targetList = (List *) projInfo->pi_state.expr;
+     int  numSimpleVars = 0;
+
+  int numNonSimpleVars = 0;
+
+    // TODO Re-evaluate this logic and may be we don't need numNonSimpleVars
+   for (int i = 0; i < projInfo->pi_state.steps_len; i++)
+     {
+         ExprEvalStep *step = &projInfo->pi_state.steps[i];
+         ExprEvalOp opcode = ExecEvalStepOp(&projInfo->pi_state, step);
+             if ( opcode == EEOP_ASSIGN_INNER_VAR ||
+                     opcode == EEOP_ASSIGN_OUTER_VAR ||
+                     opcode == EEOP_ASSIGN_SCAN_VAR)
+                     numSimpleVars++;
+             else if (opcode == EEOP_ASSIGN_TMP_MAKE_RO ||
+                        opcode == EEOP_ASSIGN_TMP)
+                     numNonSimpleVars++;
```

Comment:
It's a little hard to see what's being done in each branch of the `if`-statement because of the line-breaks in the condition so I would suggest we use `{}` even though its not strictly needed.
```suggestion
        if (opcode == EEOP_ASSIGN_INNER_VAR ||
            opcode == EEOP_ASSIGN_OUTER_VAR ||
            opcode == EEOP_ASSIGN_SCAN_VAR)
        {
            numSimpleVars++;
        }
        else if (opcode == EEOP_ASSIGN_TMP_MAKE_RO ||
            opcode == EEOP_ASSIGN_TMP)
        {
            numNonSimpleVars++;
        }
```